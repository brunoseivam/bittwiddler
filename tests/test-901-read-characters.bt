# Input data generated with
# ./gen_bin_data u8 2 s "Barry" u16 10 s "Ann" u16 55

func read_character():None {
    var name : string;
    var level : uint16;

    emit("  \"{name}\": {\"level\": {level} }\n");
}

main {
    var num_chars : uint8;

    emit("{\n");
    var i : uint8 = 0;
    while i < num_chars {
        read_character();
        i = i + 1;
    }
    emit("}\n");
}
