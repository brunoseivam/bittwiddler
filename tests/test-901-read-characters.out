{
  "Barry": {"level": 10 }
  "Ann": {"level": 55 }
}
